core/src/main/scala/akka/persistence/cassandra/query/CassandraReadJournalProvider.scala

```diff
@@ -11,10 +11,12 @@ import com.typesafe.config.Config
 class CassandraReadJournalProvider(system: ExtendedActorSystem, config: Config, configPath: String)
     extends ReadJournalProvider {
 
-  override val scaladslReadJournal: scaladsl.CassandraReadJournal =
-    new scaladsl.CassandraReadJournal(system, config, configPath)
+  private lazy val scaladslReadJournalInstance = new scaladsl.CassandraReadJournal(system, config, configPath)
```

Comment:
Better make it non lazy and keep current instantiation on construction?

Reply:
Interesting. Since the abstract `scaladslReadJournal` as overridden as a `val`, I assumed it would keep its lazy nature after compilation. Will change. 

I think the same applies [here](https://github.com/akka/akka-persistence-r2dbc/blob/main/core/src/main/scala/akka/persistence/r2dbc/query/R2dbcReadJournalProvider.scala), should I change that to keep the semantics there as well? This was introduced during the scala3 work for r2dbc.

Reply:
I PR:d such a change in https://github.com/akka/akka-persistence-r2dbc/pull/362